src/azure-cli/azure/cli/command_modules/appconfig/_params.py

```diff
@@ -189,6 +189,7 @@ def load_arguments(self, _):
 
     with self.argument_context('appconfig kv export', arg_group='AppService') as c:
         c.argument('appservice_account', validator=validate_appservice_name_or_id, help='ARM ID for AppService OR the name of the AppService, assuming it is in the same subscription and resource group as the App Configuration. Required for AppService arguments')
+        c.argument('export_as_reference', options_list=['--export-as-reference', '-r'], help="Export key-values as App Configuration references (Preview). For more information, see https://docs.microsoft.com/en-us/azure/app-service/app-service-configuration-references ")
```

Comment:
We can add preview tag to this argument (no need to add 'Preview' in the help text).

```
c.argument('export_as_reference', is_preview=True, .....)
```

Reply:
Another thing we can add is the three state flag:
` c.argument('export_as_reference', is_preview=True, arg_type=get_three_state_flag(), .......)`

This enables users to provide true/false/empty values to this argument. You can look at `--enable-public-network` argument for sample usage.

Reply:
Can you also add a `validator` for this argument?
I think we should throw an error if this argument is used with any destination other than AppService. You can look at this validator for a sample: https://github.com/Azure/azure-cli/blob/f7479671e841d196d3312523c302bab853eb25c2/src/azure-cli/azure/cli/command_modules/appconfig/_validators.py#L226

Reply:
Update pushed for this.